src/operator/controllers/metadata/annotations.go

```diff
@@ -10,20 +10,12 @@ const (
 	TLSSecretNameAnnotation           = "credentials-operator.otterize.com/tls-secret-name"
 	TLSSecretNameAnnotationDeprecated = "spire-integration.otterize.com/tls-secret-name"
 
-	// ServiceAccountAWSRoleARNAnnotation is used by EKS (Kubernetes at AWS) to link between service accounts
-	// and IAM roles
-	ServiceAccountAWSRoleARNAnnotation = "eks.amazonaws.com/role-arn"
-
 	// IAMRoleFinalizer indicates that cleanup on IAM roles is needed upon termination.
 	IAMRoleFinalizer = "credentials-operator.otterize.com/iam-role"
 
 	// AWSRoleFinalizer indicates that cleanup on AWS is needed upon termination.
 	AWSRoleFinalizer = "credentials-operator.otterize.com/aws-role"
```

Comment:
TODO: double-check that our new code handles old pods & services marked with this finalizer